src/components/MediaPlayer/VideoJS/VideoJSPlayer.js

```diff
@@ -144,6 +147,24 @@ function VideoJSPlayer({
       player.getChild('controlBar').qualitySelector.setIcon('cog');
     });
 
+    player.on('emptied', () => {
+      /**
+       * In the quality-selector plugin used in Ramp, when the player is using remote 
+       * text tracks they get cleared upon quality selection.
+       * This is a known issue with @silvermine/videojs-quality-selector plugin.
+       * When a new source is selected this event is invoked. So, we are using this event
+       * to check whether the current video player has tracks when tracks are available and
+       * adding them back in.
+       */
+      if (tracksRef.current?.length > 0 && isVideo
+        && player.textTracks()?.length <= tracksRef.current?.length) {
+        if (tracksRef.current?.length > 0 && isVideo) {
+          tracksRef.current.forEach(function (track) {
+            player.addRemoteTextTrack(track, false);
```

Comment:
Does this need to de-dupe the tracks when adding?

Reply:
I tested this again in Safari, and it seems removing and adding them back in is a robust solution as Safari handles tracks different than Firefox and Chrome. As the check `player.textTracks()?.length <= tracksRef.current?.length` fails in Safari causing the tracks to duplicate.
Thanks for catching that 🎉